### PR TITLE
Cherrypick to update release for 0.6

### PR DESCRIPTION
**Pull request by shauheen**
Cherrypick into release for RC2 0.6

### CODE REVIEW

File: src/Microsoft.ML.Core/Utilities/Contracts.cs [hunk 1]
```diff
@@ -570,7 +570,14 @@ public static void CheckParamValue<T>(this IExceptionContext ctx, bool f, T valu
...
 
-    public static void CheckValue<T>(T val, string paramName) where T : class
+    public static T CheckRef<T>(this IExceptionContext ctx, T val, string paramName, string msg) where T : class
+    {
+        if (object.ReferenceEquals(val, null))
+            throw ExceptValue(ctx, paramName, msg);
+        return val;
+    }
+
+        public static void CheckValue<T>(T val, string paramName) where T : class
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.Core/Utilities/Contracts.cs]: >     [](start = 4, length = 4)

spacing

File: src/Microsoft.ML.StandardLearners/Standard/LogisticRegression/LogisticRegression.cs [hunk 2]
```diff
@@ -54,10 +54,24 @@ public sealed class Arguments : ArgumentsBase
         /// <param name="labelColumn">The name of the label column.</param>
         /// <param name="featureColumn">The name of the feature column.</param>
         /// <param name="weightColumn">The name for the example weight column.</param>
+        /// <param name="enforceNoNegativity">Enforce non-negative weights.</param>
+        /// <param name="l1Weight">Weight of L1 regularizer term.</param>
```
Comment (artidoro) [inline src/Microsoft.ML.StandardLearners/Standard/LogisticRegression/LogisticRegression.cs]: In most places the comment says "regularization" I think "regularizer" is the naming of an internal object.

File: src/Microsoft.ML.StandardLearners/Standard/LogisticRegression/MulticlassLogisticRegression.cs [hunk 3]
```diff
@@ -76,10 +76,22 @@ public sealed class Arguments : ArgumentsBase
         /// <param name="labelColumn">The name of the label column.</param>
         /// <param name="featureColumn">The name of the feature column.</param>
         /// <param name="weightColumn">The name for the example weight column.</param>
+        /// <param name="enforceNoNegativity">Enforce non-negative weights.</param>
+        /// <param name="l1Weight">Weight of L1 regularizer term.</param>
```
Comment (artidoro) [inline src/Microsoft.ML.StandardLearners/Standard/LogisticRegression/MulticlassLogisticRegression.cs]: >regularizer [](start = 48, length = 11)

If possible switch to "regularization".

File: src/Microsoft.ML.StandardLearners/Standard/PoissonRegression/PoissonRegression.cs [hunk 4]
```diff
@@ -46,10 +46,22 @@ public sealed class Arguments : ArgumentsBase
         /// <param name="labelColumn">The name of the label column.</param>
         /// <param name="featureColumn">The name of the feature column.</param>
         /// <param name="weightColumn">The name for the example weight column.</param>
+        /// <param name="enforceNoNegativity">Enforce non-negative weights.</param>
+        /// <param name="l1Weight">Weight of L1 regularizer term.</param>
```
Comment (artidoro) [inline src/Microsoft.ML.StandardLearners/Standard/PoissonRegression/PoissonRegression.cs]: If possible, again, switch to "regularization".

**Comment by shauheen** [comment]
Thanks @artidoro and @Ivanidzo4ka, since this is just a cherry pick those comments will need to be addressed in separate PRs, and hopefully in 0.7. Thanks for the reviews.

**Pull request merged**